File: backend/src/main/java/br/gov/mt/seplag/sgd/service/StatsService.java
package br.gov.mt.seplag.sgd.service;

import br.gov.mt.seplag.sgd.dto.StatsDTO;
import br.gov.mt.seplag.sgd.repository.AlbumRepository;
import br.gov.mt.seplag.sgd.repository.ArtistRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

@Service
@RequiredArgsConstructor
public class StatsService {

    private final AlbumRepository albumRepository;
    private final ArtistRepository artistRepository;

    public StatsDTO getStats() {
        long totalArtists = artistRepository.count();
        long totalAlbums = albumRepository.count();
        
        return new StatsDTO(totalArtists, totalAlbums);
    }
}

File: frontend/src/services/statsService.ts
import apiClient from './apiClient';

export interface StatsDTO {
  totalArtists: number;
  totalAlbums: number;
}

export const statsService = {
  getStats: async (): Promise<StatsDTO> => {
    const response = await apiClient.get<StatsDTO>('/api/v1/stats');
    return response.data;
  }
};
